src/recordSchema.ts

```diff
@@ -0,0 +1,63 @@
+
+import { Dictionary, expandUrl, loadField, idmapFieldsUnionOfNoneTypeOrArrayOfRecordFieldLoader, typeDSLEnumd9cba076fca539106791a4f46d198c7fcfbdb779Loader2, LoadingOptions, Saveable, ValidationException } from './util/internal'
+
+export class RecordSchema extends Saveable {
+  fields?: any
+  type: any
+  extensionFields?: Dictionary<any>
+  loadingOptions?: LoadingOptions
+
+  constructor (type: any, fields?: any, extensionFields?: Dictionary<any>, loadingOptions?: LoadingOptions) {
+    super()
+    this.extensionFields = extensionFields ?? {}
+    this.loadingOptions = loadingOptions ?? new LoadingOptions({})
+    this.fields = fields
+    this.type = type
+  }
+
+  static override async fromDoc (doc: any, baseuri: string, loadingOptions: LoadingOptions, docRoot?: string): Promise<Saveable> {
+    const _doc = Object.assign({}, doc)
+    const errors: ValidationException[] = []
+    if ('fields' in _doc) {
+      try {
+        var fields = await loadField(_doc.fields, idmapFieldsUnionOfNoneTypeOrArrayOfRecordFieldLoader, baseuri, loadingOptions)
```

Comment:
@ZimmerA should we use this `fields` value, or was it instantiated here just to test that the field can be loaded?

Reply:
It is used later on in the RecordSchema constructor. Since it is declared as `var` it has function scope, so it can be used outside the scope it is declared in within that function.

But I agree, the style is a bit weird... I already changed this in the codegen, will fix it here too 👍🏻